From bd525f93a10fb31a638b795b3edf770d57eda06c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Muhammed=20Efe=20=C3=87etin?= <efectn@protonmail.com>
Date: Fri, 27 Jan 2023 17:54:51 +0300
Subject: [PATCH] Import
 VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver on wireless drivers

---
 drivers/net/wireless/rtl8188eu/os_dep/osdep_service.c | 3 +++
 drivers/net/wireless/rtl8189es/os_dep/osdep_service.c | 3 +++
 drivers/net/wireless/rtl8189fs/os_dep/osdep_service.c | 3 +++
 drivers/net/wireless/rtl8192eu/os_dep/osdep_service.c | 4 ++++
 drivers/net/wireless/rtl8723ds/os_dep/osdep_service.c | 4 ++++
 drivers/net/wireless/rtl8723du/os_dep/osdep_service.c | 4 ++++
 drivers/net/wireless/rtl8811cu/os_dep/osdep_service.c | 3 +++
 drivers/net/wireless/rtl8822bs/os_dep/osdep_service.c | 4 ++++
 drivers/net/wireless/rtl88x2bu/os_dep/osdep_service.c | 3 +++
 drivers/net/wireless/rtl88x2cs/os_dep/osdep_service.c | 3 +++
 10 files changed, 34 insertions(+)

diff --git a/drivers/net/wireless/rtl8188eu/os_dep/osdep_service.c b/drivers/net/wireless/rtl8188eu/os_dep/osdep_service.c
index c6b3f65bb..38bfd52cc 100644
--- a/drivers/net/wireless/rtl8188eu/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8188eu/os_dep/osdep_service.c
@@ -27,6 +27,9 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 
 #if defined(PLATFORM_LINUX)
 /*
diff --git a/drivers/net/wireless/rtl8189es/os_dep/osdep_service.c b/drivers/net/wireless/rtl8189es/os_dep/osdep_service.c
index d6d326aef..aa1f0499d 100644
--- a/drivers/net/wireless/rtl8189es/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8189es/os_dep/osdep_service.c
@@ -27,6 +27,9 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 
 #if defined(PLATFORM_LINUX)
 /*
diff --git a/drivers/net/wireless/rtl8189fs/os_dep/osdep_service.c b/drivers/net/wireless/rtl8189fs/os_dep/osdep_service.c
index 04000c768..d7005b9dc 100644
--- a/drivers/net/wireless/rtl8189fs/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8189fs/os_dep/osdep_service.c
@@ -27,6 +27,9 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 
 #if defined(PLATFORM_LINUX)
 /*
diff --git a/drivers/net/wireless/rtl8192eu/os_dep/osdep_service.c b/drivers/net/wireless/rtl8192eu/os_dep/osdep_service.c
index e6954c36f..370788123 100644
--- a/drivers/net/wireless/rtl8192eu/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8192eu/os_dep/osdep_service.c
@@ -20,6 +20,10 @@
 
 #define RT_TAG	'1178'
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
+
 #if defined(PLATFORM_LINUX)
 /*
 * Translate the OS dependent @param error_code to OS independent RTW_STATUS_CODE
diff --git a/drivers/net/wireless/rtl8723ds/os_dep/osdep_service.c b/drivers/net/wireless/rtl8723ds/os_dep/osdep_service.c
index cb5ff6940..6c496f348 100644
--- a/drivers/net/wireless/rtl8723ds/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8723ds/os_dep/osdep_service.c
@@ -30,6 +30,10 @@ atomic_t _malloc_cnt = ATOMIC_INIT(0);
 atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
+
 /*
 * Translate the OS dependent @param error_code to OS independent RTW_STATUS_CODE
 * @return: one of RTW_STATUS_CODE
diff --git a/drivers/net/wireless/rtl8723du/os_dep/osdep_service.c b/drivers/net/wireless/rtl8723du/os_dep/osdep_service.c
index ed9c912cb..a4686dffa 100644
--- a/drivers/net/wireless/rtl8723du/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8723du/os_dep/osdep_service.c
@@ -7,6 +7,10 @@
 
 #define RT_TAG	'1178'
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
+
 /*
 * Translate the OS dependent @param error_code to OS independent RTW_STATUS_CODE
 * @return: one of RTW_STATUS_CODE
diff --git a/drivers/net/wireless/rtl8811cu/os_dep/osdep_service.c b/drivers/net/wireless/rtl8811cu/os_dep/osdep_service.c
index f9f494d8c..62e664b78 100755
--- a/drivers/net/wireless/rtl8811cu/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8811cu/os_dep/osdep_service.c
@@ -27,6 +27,9 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 
 #if defined(PLATFORM_LINUX)
 /*
diff --git a/drivers/net/wireless/rtl8822bs/os_dep/osdep_service.c b/drivers/net/wireless/rtl8822bs/os_dep/osdep_service.c
index c32af34f2..86d0257c4 100644
--- a/drivers/net/wireless/rtl8822bs/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8822bs/os_dep/osdep_service.c
@@ -23,6 +23,10 @@
 #define kthread_complete_and_exit(comp, code) complete_and_exit(comp, code)
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
+
 #define RT_TAG	'1178'
 
 #ifdef DBG_MEMORY_LEAK
diff --git a/drivers/net/wireless/rtl88x2bu/os_dep/osdep_service.c b/drivers/net/wireless/rtl88x2bu/os_dep/osdep_service.c
index 3b927f4fb..7f0f72ea8 100644
--- a/drivers/net/wireless/rtl88x2bu/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl88x2bu/os_dep/osdep_service.c
@@ -27,6 +27,9 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 
 #if defined(PLATFORM_LINUX)
 /*
diff --git a/drivers/net/wireless/rtl88x2cs/os_dep/osdep_service.c b/drivers/net/wireless/rtl88x2cs/os_dep/osdep_service.c
index 883d4edd3..88476ec3d 100644
--- a/drivers/net/wireless/rtl88x2cs/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl88x2cs/os_dep/osdep_service.c
@@ -27,6 +27,9 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+#endif
 
 #if defined(PLATFORM_LINUX)
 /*
-- 
2.39.1

